### ISSUE TITLE
Questions about the summary tax tables

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#1325]
As part of developing the user documentation, I've been adding descriptions of several calculated output variables.  In the course of doing this, I've noticed some inconsistencies in the aggregate tables prepared by the Tax-Calculator (see, for example, the `create_distribution_table` utility function) and in the tables shown on the TaxBrain static results page.

Here is a TaxBrain results table for [run 11490](http://www.ospc.org/taxbrain/11490/):

![image](https://cloud.githubusercontent.com/assets/12170745/25550155/d8f9bec4-2c44-11e7-998b-5b5c9f8f6275.png)

The problem is that the numbers seem inconsistent as you move across the columns.  Focusing on the `All` row, we see two unexpected results.

First, the `Regular Tax` and the `AMT` amounts don't add up to the `Tax before Credits` amount --- most people would expect that sum to be equal to the `Tax before Credits` amount.

Second, when the `Non-refundable Credits` amount is subtracted from the `Tax before Credits` amount, the difference is not equal to the `Tax before Refundable Credits` amount --- most people would expect that difference to be equal to the `Tax before Refundable Credits` amount.

These inconsistencies are caused by using `c05200` rather than `taxbc` for the `Regular Tax` amount and by using `c07100` rather than `(c05800 - c09200)` for the `Non-refundable Credits` amount.  The latest version of the [user documentation](http://open-source-economics.github.io/Tax-Calculator/#output) clarifies the difference between the variables used now and the variables that seem more sensible.

I'd be happy to prepare a pull request that makes these changes, but I'm hoping for some kind of discussion here before I start doing that.

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @GoFroggyRun @codykallen @zrisher 

P.S.  When @GoFroggyRun wrote the [`ODD_RECORDS.md`](https://github.com/open-source-economics/Tax-Calculator/blob/master/ODD_RECORDS.md) documentation many months ago, his discussion made it clear that `taxbc` (not `c05200`) represents `Regular Tax`.

**Comment by MattHJensen** [comment]
@martinholmer, your assessment of the problem and solution makes sense to me. Very happy you found this.

**Comment by martinholmer** [comment]
After further investigation, my analysis of the situation regarding non-refundable tax credits was wrong in issue #1325.  The `c07100` variable has always been total non-refundable credits used to reduce positive tax liability (which might be less than the total potential non-refundable credits).  

But the relationship between tax before refundable credits, `c09200`, and `c05800` is that `c09200` but not `c05800` includes `othertaxes`, which we do not have in the summary tables.  I'll change all the user documentation and add a test to verify this new understanding, but the items in the summary table still won't "add up" because we don't have the option to display `othertaxes`.   The `othertaxes` calculated variable includes mostly the NIIT and penalties for early withdrawals from qualified retirement plans (e.g., 401(k) and IRA plans).

I will add `othertaxes` to the diagnostic table produced by Tax-Calculator in a forthcoming pull request.

We should probably discuss the pros and cons of adding `othertaxes` to the TaxBrain distribution tables (if we really want the columns to "add up").

@MattHJensen @feenberg @Amy-Xu @andersonfrailey @GoFroggyRun @codykallen @zrisher

**Comment by feenberg** [comment]
On Sun, 30 Apr 2017, Martin Holmer wrote:
> 
> We should probably discuss the pros and cons of adding othertaxes to the
> TaxBrain distribution tables (if we really want the columns to "add up").
>

The NIIT is of particular interest, since it affect the marginal tax rate. 
So we need to include it in the total. It may not require a separate 
column, but it would be nice to make that available.

dan

**Comment by martinholmer** [comment]
@feenberg said:
> The NIIT is of particular interest, since it affects the marginal tax rate.
> So we need to include it in the total. It may not require a separate column, but it would be nice to make that available.

I agree.  Perhaps the TaxBrain distribution table could look more like the Tax-Calculator diagnostic table.  Here is what TaxBrain looks like now.

![image](https://cloud.githubusercontent.com/assets/12170745/25550155/d8f9bec4-2c44-11e7-998b-5b5c9f8f6275.png)

Maybe, we could replace the current sequence:
```
Tax before Credits
Non-refundable Credits
Tax before Refundable Credits
Refundable Credits
Individual Income Tax Liabilities
```
with something closer to what is on the Form 1040:
```
(3) Tax before Credits                  <--- UNCHANGED
(4) Non-refundable Credits Used         <--- emphasize "used" rather than "potential"
(5) Other Taxes                         <--- add this new component
(6) Refundable Credits                  <--- UNCHANGED
(7) Individual Income Tax Liabilities   <--- UNCHANGED
```
With this table, `(7) = (3) - (4) + (5) - (6)`.

**Comment by martinholmer** [comment]
Issue #1325 has been resolved by pull request #1328.
